xmtp_id/src/associations/association_log.rs

```diff
@@ -381,6 +381,7 @@ impl IdentityAction for IdentityUpdate {
         }
 
         let new_state = state.ok_or(AssociationError::NotCreated)?;
+        // Do we need to handle this case for checksum inboxIds?
```

Comment:
@insipx I could hack some code here to check for mismatched inboxIds from checksums here but it's a lot less straight forward thoughts on this case?